pennylane/devices/default_qubit.py

```diff
@@ -44,6 +45,14 @@
 TPHASE = np.exp(1j * np.pi / 4)
 
 
+def deprecation_warning(func_name):
+    warnings.warn(
+        f"DefaultQubit.{func_name} is deprecated. Use qml.devices.experimental.DefaultQubit2 "
+        "instead.",
```

Comment:
I'm not sure if we're ready to recommend DefaultQubit2 as a replacement. @albi3ro?
Maybe something like:
> Directly calling the methods of default.qubit is not recommended and should be avoided where possible. The available methods of default.qubit will be updated soon to follow a new API described here: https://docs.pennylane.ai/en/stable/code/api/pennylane.devices.experimental.DefaultQubit2.html

Reply:
I've added filters to `QubitDevice.execute`, `QubitDevice.batch_execute`, `Device.check_validity`, so the warnings won't show up in user facing workflows for executing qnodes/tapes. In all other cases, the warning will show up. So in most cases, the warning will only show up when a user directly accesses the functions being deprecated.